### ISSUE TITLE
healthcheck is not able to mark instances as "unhealthy" when host on which the instance is running is stopped. 

### ISSUE THREAD

**Issue by sangeethah** [rancherio/rancher#1014]
Server Version - rancher/server:v0.23.0-rc1

Steps to reproduce the problem:
Set up has 3 hosts:

Create a service with scale 3 and health check enabled.
In my case , I also use the following labels for containers within the service to be hosted in different hosts - io.rancher.scheduler.affinity:container_label_ne = io.rancher.service.name=

Activate the service.
Service gets activated successfully and 3 containers are deployed in 3 different containers.

Stopped one of the host where "test_testme_2" instance is running.

This results in 2 of the hosts marking the state of this instance as "unhealthy".
But the host that was down reports the state of this instance as being  "healthy"

I dont see the state of the instance being marked "unhealthy" in instance table for replication to happen.

mysql>select health_state from healthcheck_instance_host_map where healthcheck_instance_id=(select id from healthcheck_instance where instance_id=9);
+--------------+---------+
| health_state | host_id |
+--------------+---------+
| unhealthy    |       1 |
| unhealthy    |       2 |
| healthy      |       3 |
+--------------+---------+
3 rows in set (0.00 sec)

mysql> select id,name,state,health_state from instance where name like "%testme%";              
+----+---------------+---------+--------------+
| id | name          | state   | health_state |
+----+---------------+---------+--------------+
|  8 | test_testme_1 | running | healthy      |
|  9 | test_testme_2 | running | healthy      |
| 10 | test_testme_3 | purged  | healthy      |
| 11 | test_testme_4 | purged  | healthy      |
| 12 | test_testme_4 | purged  | healthy      |
| 13 | test_testme_3 | running | healthy      |
+----+---------------+---------+--------------+
6 rows in set (0.00 sec)

**Comment by alena1108** [comment]
Clarification - by "Stopping the host" Sangeetha means bringing down the container where rancher agent is running.

**Comment by alena1108** [comment]
Unfortunately down host is not reflected as down in the DB due to https://github.com/rancherio/rancher/issues/701. So in healthcheck code we can't remove the host from monitoring as to it still presented as Active

**Comment by alena1108** [comment]
@ibuildthecloud I've put in a PR for the host_map to get removed on the host.remove event. I wonder if fix for #701 should fix this bug as well?

**Comment by sangeethah** [comment]
Reopening this issue.
Tested with server version - v0.24.0-rc1

Had a server with scale 5 where 3 instances of the service was running on host1.
Stopped host1.

I see the host is in "reconnecting" state and all 3 instances were identified as being unhealthy.
For 2 of the instances , we were able to deploy 2 more instances in other hosts.
But 1 of instances was attempted to start in the host that is in "reconnecting" state.

Closing this issue.
